chain/ethereum/ethereum.go

```diff
@@ -208,3 +210,25 @@ func Backfill(ctx context.Context, client *ethclient.Client, addresses []common.
 	}
 	return nil
 }
+
+// BackfillFrom queries past blocks for the events emitted by the given contract addresses and feeds these events into the event log chan.
+// * fromBlock > 0 && numBlocks > 0 => Backfill from fromBlock to fromBlock+numBlocks
+// * fromBlock > 0 && numBlocks = 0 => Backfill from fromBlock to current latest block
+// * fromBlock = 0 && numBlocks > 0 => Backfill last numBlocks blocks
+func BackfillFrom(ctx context.Context, client *ethclient.Client, addresses []common.Address, logs chan types.Log, fromBlock uint64, numBlocks uint64) error {
+	switch {
+	case fromBlock > 0 && numBlocks > 0:
+		return Backfill(ctx, client, addresses, logs, fromBlock, fromBlock+numBlocks)
+	case fromBlock > 0 && numBlocks == 0:
+		return Backfill(ctx, client, addresses, logs, fromBlock, 0)
+	case fromBlock == 0 && numBlocks > 0:
+		toBlock, err := client.BlockNumber(ctx)
+		if err != nil {
+			log.Error().Err(err).Msg("failed to get block number")
+			return err
+		}
+		return Backfill(ctx, client, addresses, logs, toBlock-numBlocks, toBlock)
```

Comment:
please replace `toBlock` with `0`

Reply:
replace toBlock with 0 will cause the Backfill function to make rpc to get the latest block again